seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/excel/ExcelReaderListener.java

```diff
@@ -0,0 +1,277 @@
+
+    private SeaTunnelDataType<?>[] fieldTypes;
+
+    Map<Integer, String> customHeaders = new HashMap<>();
+
+    public ExcelReaderListener(
+            String tableId,
+            Collector<SeaTunnelRow> output,
+            Config pluginConfig,
+            SeaTunnelRowType seaTunnelRowType) {
+        this.tableId = tableId;
+        this.output = output;
+        this.pluginConfig = pluginConfig;
+        this.seaTunnelRowType = seaTunnelRowType;
+
+        fieldTypes = seaTunnelRowType.getFieldTypes();
+
+        if (pluginConfig.hasPath(BaseSourceConfigOptions.DATE_FORMAT.key())) {
+            String dateFormatString =
+                    pluginConfig.getString(BaseSourceConfigOptions.DATE_FORMAT.key());
+            dateFormatter = DateTimeFormatter.ofPattern(dateFormatString);
+        }
+        if (pluginConfig.hasPath(BaseSourceConfigOptions.DATETIME_FORMAT.key())) {
+            String datetimeFormatString =
+                    pluginConfig.getString(BaseSourceConfigOptions.DATETIME_FORMAT.key());
+            dateTimeFormatter = DateTimeFormatter.ofPattern(datetimeFormatString);
+        }
+        if (pluginConfig.hasPath(BaseSourceConfigOptions.TIME_FORMAT.key())) {
+            String timeFormatString =
+                    pluginConfig.getString(BaseSourceConfigOptions.TIME_FORMAT.key());
+            timeFormatter = DateTimeFormatter.ofPattern(timeFormatString);
+        }
+    }
+
+    @Override
+    public void invokeHead(Map<Integer, ReadCellData<?>> headMap, AnalysisContext context) {
+        for (int i = 0; i < headMap.size(); i++) {
+            String header = headMap.get(i).getStringValue();
+            if (!"null".equals(header)) {
+                customHeaders.put(i, header);
+            }
+        }
+    }
+
+    @Override
+    public void invoke(Map<Integer, Object> data, AnalysisContext context) {
+        cellCount = data.size();
+        SeaTunnelRow seaTunnelRow = new SeaTunnelRow(fieldTypes.length);
+        Map<Integer, Cell> cellMap = context.readRowHolder().getCellMap();
+        int i = 0;
+        for (; i < fieldTypes.length; i++) {
+            if (cellMap.get(i) == null) {
+                seaTunnelRow.setField(i, null);
+            } else {
+                Object cell = convert(data.get(i), cellMap.get(i), fieldTypes[i]);
+                seaTunnelRow.setField(i, cell);
+            }
+        }
+        seaTunnelRow.setTableId(tableId);
+        output.collect(seaTunnelRow);
+    }
+
+    @Override
+    public void doAfterAllAnalysed(AnalysisContext context) {
+        log.info("excel parsing completed");
+    }
+
+    @Override
+    public void onException(Exception exception, AnalysisContext context) {
+        log.error("cell parsing exception :{}", exception.getMessage());
+        if (exception instanceof ExcelDataConvertException) {
+            ExcelDataConvertException excelDataConvertException =
+                    (ExcelDataConvertException) exception;
+            log.error(
```

Comment:
Reduce big log message.
```suggestion
            log.debug(
```